### PR TITLE
PWN-7030 - finance block without sample

### PR DESCRIPTION
**Pull request by NikolaiEmelianov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7030

## Description of Work

finance block cell

### CODE REVIEW

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/finance_block/UiKitFinanceBlockView.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    private val binding = inflateViewBinding<WidgetFinanceBlockBinding>()
+
+    val leftSideView: UiKitLeftSideView
+        get() = binding.leftSideView
+    val rightSideView: UiKitRightSideView
+        get() = binding.rightSideView
+
+    private var _item: FinanceBlockCellModel? = null
+    val item: FinanceBlockCellModel
+        get() = _item ?: throw IllegalStateException("Not bind yet")
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/finance_block/UiKitFinanceBlockView.kt:29]:
```suggestion
        get() = _item ?: error("Not bind yet")
```

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/icon_wrapper/UiKitIconWrapper.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    context: Context,
+    attrs: AttributeSet? = null
+) : ConstraintLayout(context, attrs) {
+
+    private var currentModel: IconWrapperCellModel? = null
+    private val viewPool = ComponentViewPool<IconWrapperCellModel>(this) {
+        when (this) {
+            IconWrapperCellModel.SingleIcon::class -> inflateViewBinding<WidgetIconWrapperSingleBinding>()
+            IconWrapperCellModel.TwoIcon::class -> inflateViewBinding<WidgetIconWrapperTwoBinding>()
+            else -> throw IllegalArgumentException("add remaining type")
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/icon_wrapper/UiKitIconWrapper.kt:22]: add remaining type -> No type for viewPool: $this

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/right_side/UiKitRightSideView.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    private val viewPool = ComponentViewPool<RightSideCellModel>(this) {
+        when (this) {
+            RightSideCellModel.TwoLineText::class -> inflateViewBinding<WidgetRightSideDoubleTextBinding>()
+            RightSideCellModel.SingleTextTwoIcon::class -> inflateViewBinding<WidgetRightSideSingleTextTwoIconBinding>()
+            else -> throw IllegalArgumentException("add remaining type")
+        }
+    }
+
+    val item: RightSideCellModel
+        get() = currentModel ?: throw IllegalStateException("Not bind yet")
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/right_side/UiKitRightSideView.kt:28]:
```suggestion
        get() = currentModel ?: error("Model is null. Maybe bind is not called?")
```

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/right_side/UiKitRightSideView.kt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    private var currentModel: RightSideCellModel? = null
+    private val viewPool = ComponentViewPool<RightSideCellModel>(this) {
+        when (this) {
+            RightSideCellModel.TwoLineText::class -> inflateViewBinding<WidgetRightSideDoubleTextBinding>()
+            RightSideCellModel.SingleTextTwoIcon::class -> inflateViewBinding<WidgetRightSideSingleTextTwoIconBinding>()
+            else -> throw IllegalArgumentException("add remaining type")
+        }
+    }
+
+    val item: RightSideCellModel
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/right_side/UiKitRightSideView.kt:27]: item -> model ?

File: ui-kit/src/main/kotlin/org/p2p/uikit/utils/drawable/shape/ShapeAppearanceExtensions.kt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            clipToOutline = true
+        }
+    }
+}
+
+fun shapeRectangle(): ShapeAppearanceModel = ShapeAppearanceModel.builder().build()
+
+fun shapeCircle(): ShapeAppearanceModel = ShapeAppearanceModel.builder()
+    .setAllCornerSizes(PILL)
+    .build()
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/utils/drawable/shape/ShapeAppearanceExtensions.kt:45]:
```suggestion
fun shapeCircle(): ShapeAppearanceModel = 
    ShapeAppearanceModel.builder()
        .setAllCornerSizes(PILL)
        .build()
```

File: ui-kit/src/main/kotlin/org/p2p/uikit/utils/drawable/DrawableCellModel.kt [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+data class DrawableCellModel(
+    val drawable: Drawable = shapeDrawable(),
+    @ColorRes val tint: Int = android.R.color.transparent,
+    // only MaterialShapeDrawable support
+    @Px val strokeWidth: Float = 0f,
+    // only MaterialShapeDrawable support
+    @ColorRes val strokeColor: Int = android.R.color.transparent,
+)
+
+fun shapeDrawable(
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/utils/drawable/DrawableCellModel.kt:21]: explicity return type

File: ui-kit/src/main/kotlin/org/p2p/uikit/utils/viewpool/ComponentViewPool.kt [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        viewGroup.removeAllViews()
+        return pair
+            ?.also { it.second.forEach { view -> viewGroup.addView(view) } }
+            ?: inflateAndSave(newModel)
+    }
+
+    private fun inflateAndSave(type: KClass<out CellModel>): BindingWithViews {
+        val binding = inflater(type)
+        val newView = mutableListOf<View>()
+        viewGroup.forEach { newView.add(it) }
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/utils/viewpool/ComponentViewPool.kt:42]: maybe we can use map?

**Review by gslevinkov — APPROVED** [review]

**Review by 1sakov — APPROVED** [review]

**Pull request merged**